src/test/javascript/spec/component/orion/orion-code-editor-instructor-and-editor-container.component.spec.ts

```diff
@@ -57,41 +61,46 @@ describe('CodeEditorInstructorAndEditorOrionContainerComponent', () => {
                 orionConnectorService = TestBed.inject(OrionConnectorService);
             });
     });
+
     afterEach(() => {
-        sinon.restore();
+        jest.restoreAllMocks();
     });
 
     it('applyDomainChange should inform connector', () => {
-        const selectRepositorySpy = spy(orionConnectorService, 'selectRepository');
+        const selectRepositorySpy = jest.spyOn(orionConnectorService, 'selectRepository');
 
         // @ts-ignore
         comp.applyDomainChange({}, {});
 
-        expect(selectRepositorySpy).to.have.been.calledOnceWithExactly(REPOSITORY.TEST);
+        expect(selectRepositorySpy).toHaveBeenCalledTimes(1);
+        expect(selectRepositorySpy).toHaveBeenCalledWith(REPOSITORY.TEST);
     });
     it('ngOnInit should subscribe to orionState', () => {
-        const orionStateStub = stub(orionConnectorService, 'state');
+        const orionStateStub = jest.spyOn(orionConnectorService, 'state');
         const orionState = { opened: 40, building: false, cloning: false } as any;
-        orionStateStub.returns(new BehaviorSubject(orionState));
+        orionStateStub.mockReturnValue(new BehaviorSubject(orionState));
 
         comp.ngOnInit();
 
-        expect(orionStateStub).to.have.been.calledOnceWithExactly();
-        expect(comp.orionState).to.be.deep.equals(orionState);
+        expect(orionStateStub).toHaveBeenCalledTimes(1);
+        expect(orionStateStub).toHaveBeenCalledWith();
+        expect(comp.orionState).toEqual(orionState);
     });
     it('buildLocally should call connector', () => {
-        const buildLocallySpy = spy(orionConnectorService, 'buildAndTestLocally');
-        const isBuildingSpy = spy(orionConnectorService, 'isBuilding');
+        const buildLocallySpy = jest.spyOn(orionConnectorService, 'buildAndTestLocally');
+        const isBuildingSpy = jest.spyOn(orionConnectorService, 'isBuilding');
 
         comp.buildLocally();
 
-        expect(isBuildingSpy).to.have.been.calledOnceWithExactly(true);
-        expect(buildLocallySpy).to.have.been.calledOnceWithExactly();
+        expect(isBuildingSpy).toHaveBeenCalledTimes(1);
+        expect(isBuildingSpy).toHaveBeenCalledWith(true);
+        expect(buildLocallySpy).toHaveBeenCalledTimes(1);
+        expect(buildLocallySpy).toHaveBeenCalledWith();
     });
     it('submit should call connector', () => {
-        const submitSpy = spy(orionConnectorService, 'submit');
-        const isBuildingSpy = spy(orionConnectorService, 'isBuilding');
-        const listenOnBuildOutputSpy = spy(TestBed.inject(OrionBuildAndTestService), 'listenOnBuildOutputAndForwardChanges');
+        const submitSpy = jest.spyOn(orionConnectorService, 'submit');
+        const isBuildingSpy = jest.spyOn(orionConnectorService, 'isBuilding');
+        const listenOnBuildOutputSpy = jest.spyOn(TestBed.inject(OrionBuildAndTestService), 'listenOnBuildOutputAndForwardChanges');
```

Comment:
Even though you might only use it here, I'd suggest moving the injection up to stay consistent with all the other injections. I think it makes sense to define them not in the tests themselves.

Reply:
Interesting. Personally I dont really see the benefit of adding two more lines for nothing, but apparently almost all other tests do it that way, so I changed it. I understand that it improves the extensibility because new tests can reuse the services now, but by that logic you would also have to define all spies top level, which would end up in a giant mess of variables. So in the same way we define spies at the lowest level where they are needed, I would store services only needed for one test in there. Am I overlooking something?

Reply:
I suggest you don't resolve the conversation if you have remaining questions :D

I think it's just a convention that has developed. Probably because the Testbed configuration is done above so all injections were done above as well. But of spies, stubs, mocks this was never a thing. 🤷 

Another point you have to think about is again consistency. It makes sense to keep everything where it commonly is because then people look at that places if they search for it.